data_validation/__main__.py

```diff
@@ -314,12 +317,27 @@ def build_config_managers_from_args(args):
     return configs
 
 
-def build_config_managers_from_yaml(args):
-    """Returns List[ConfigManager] instances ready to be executed."""
-    config_managers = []
+def config_runner(args):
+    if args.config_dir:
+        mgr = state_manager.StateManager(file_system_root_path=args.config_dir)
+        config_file_names = mgr.list_validations_in_dir(args.config_dir)
 
-    config_file_path = _get_arg_config_file(args)
-    yaml_configs = _get_yaml_config_from_file(config_file_path)
+        config_managers = []
+        for file in config_file_names:
+            config_managers.extend(build_config_managers_from_yaml(args, file))
+    else:
+        config_file_path = _get_arg_config_file(args)
+        config_managers = build_config_managers_from_yaml(args, config_file_path)
+
+    run_validations(args, config_managers)
+
+
+def build_config_managers_from_yaml(args, config_file_path):
+    """Returns List[ConfigManager] instances ready to be executed."""
+    if "config_dir" in args and args.config_dir:
```

Comment:
Why do we need the first conditional? Can't we just do `if args.config_dir`

Reply:
It's because of the (still supported) `run-config` legacy option - I've only added the `config-dir` subparser to the newer `configs run`, so when `run-config` is used, the `config-dir` variable is not initialized to null and so it errors.